tools/automation/cli_linter/rules/help_rules.py

```diff
@@ -48,31 +49,36 @@ def faulty_help_example_rule(linter, help_entry):
 
 @help_file_entry_rule
 def faulty_help_example_parameters_rule(linter, help_entry):
+    target = "azure.cli.core.parser.AzCliCommandParser.error"
+    parser = linter.command_parser
     violations = []
-    valid_commands = linter._loaded_help.keys()
-    for index, example in enumerate(linter.get_help_entry_examples(help_entry)):
-        example_text = example.get('text','')
-        cmd_params_list = get_cmd_param_list_from_example(example_text)
-
-        for cmd_body, params in cmd_params_list:
-            cmd_list = cmd_body.split()[1:] # break down command into list of words and get rid of `az`.
 
-            while(cmd_list and " ".join(cmd_list) not in valid_commands): # trim possible positional arguments.
-                cmd_list = cmd_list[:-1]
-
-            # if no help entry match was found, this is not a valid command.
-            if not cmd_list:
-                violations.append("\t\t\t'{}' is not a valid command.".format(cmd_body))
-                continue
-
-            cmd = " ".join(cmd_list)
-
-            for param in params:
-                if not linter.is_valid_parameter_help_option(cmd, param):
-                    violations.append("\t\t\tOption '{}' is not a valid parameter for command '{}'.".format(param, cmd))
+    for example in linter.get_help_entry_examples(help_entry):
+        example_text = example.get('text','')
+        commands = extract_commands_from_example(example_text)
+        i = 0
+        while i < len(commands):
+            command = commands[i]
+            nested_commands = []
+
+            with mock.patch(target) as mocked_error_method:
+                mocked_error_method.side_effect = SystemExit # mock call of parser.error so usage won't be printed.
+                try:
+                    command_args = shlex.split(command)[1:]
+                    command_args, nested_commands = process_command_args(command_args)
+                    parser.parse_args(command_args)
+                except SystemExit as e:
+                    violation = "\tThe following command:\n\t\t{}\n\tis not a valid command".format(command)
```

Comment:
I don't think you need to print "The following command...." for each command found. Just the command should be fine.